infrastructure/api_stack.py

```diff
@@ -15,22 +16,27 @@
 from .roles import MAX_SESSION_DURATION
 
 
-class APIStack(Stack):
+@dataclass
```

Comment:
`NestedStackProps` isn't a dataclass, so I'm not sure this is appropriate. For example, `@dataclass` will override the `__init__` of the parent class, by default, at which point this class is not really a `NestedStackProps` anymore.

Reply:
`@dataclass` decorator is set for `APIStackProps` class which is what I wanted to do and this class behaves as I would expect from dataclass

Reply:
Could you please explain why though? Is it *necessary* to make this a dataclass?